cmd/reporting-operator/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/operate-first/curator-operator/internal"
+	"github.com/operate-first/curator-operator/pkg/operator"
+	"github.com/operate-first/curator-operator/pkg/signals"
+	"github.com/spf13/cobra"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+const (
+	defaultServerPort = ":8080"
+)
+
+var (
+	reportLog = ctrl.Log.WithName("report")
+)
+
+var (
+	cfg operator.Config
+
+	rootCmd = &cobra.Command{
+		Use:   "reporting-operator",
+		Short: "",
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return cmd.Help()
+		},
+	}
+
+	startCmd = &cobra.Command{
+		Use:   "start",
+		Short: "Starts the http server",
+		Long:  "",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := runReporting(context.Background(), reportLog)
+			if err != nil {
+				reportLog.Error(err, "error generating report")
+			}
+		},
+	}
+)
+
+func init() {
+	rootCmd.AddCommand(startCmd)
+	startCmd.Flags().StringVar(&cfg.Port, "port", defaultServerPort, "http server port")
+}
```

Comment:
I think we need to create a constructor for the HTTP server cobra.Command to avoid having to use init like this to handle CLI flag arguments:

```go
func NewCmd() *cobra.Command{
    var (
        serverPort string
    )
    cmd := &cobra.Command{
        ...
    }

    cmd.Flags().StringVar(&serverPort, "port", defaultServerPort, "configures the default port for the HTTP server")
    
    return cmd
```